language/src/ceylon/language/ArraySequence.ceylon

```diff
@@ -3,127 +3,106 @@
  Since [[Array]]s are mutable, this class is private to the
  language module, where we can be sure the `Array` is not
  modified after the `ArraySequence` has been initialized."
-by ("Tom")
+by("Tom")
 shared sealed final
 serializable
 tagged("Collections", "Sequences")
-class ArraySequence<out Element>(array)
+class ArraySequence<out Element>(Array<Element> array)
         extends Object()
         satisfies [Element+] {
     
-    Array<Element> array;
+    assert(!array.empty);
     
-    assert (!array.empty);
+    getFromFirst = array.getFromFirst;
     
-    getFromFirst(Integer index) 
-            => array.getFromFirst(index);
-    
-    contains(Object element) 
-            => array.contains(element);
+    contains = array.contains;
     
     size => array.size;
     
-    iterator() => array.iterator();
+    iterator => array.iterator;
```

Comment:
The `=>` doesn't look right. Does this compile?

Also, although `iterator = ...` does have its appeal, it's a worse way of writing this from a performance perspective. Treating `iterator` as a value results in a new instance of `Callable` and additional indirection which can affect performance.

Same for other instances of this pattern in this patch.